prow/plank/reconciler.go

```diff
@@ -894,6 +919,17 @@ func prowJobIndexer(prowJobNamespace string) ctrlruntimeclient.IndexerFunc {
 	}
 }
 
+func prowJobQueueNameIndexer(prowJobNamespace string) ctrlruntimeclient.IndexerFunc {
+	return func(o ctrlruntimeclient.Object) []string {
+		pj := o.(*prowv1.ProwJob)
+		if pj.Namespace != prowJobNamespace || pj.Spec.Agent != prowv1.KubernetesAgent || pj.Spec.JobQueueName == "" {
+			return nil
+		}
+
+		return []string{pj.Spec.JobQueueName}
```

Comment:
It is dangerous to use the raw user supplied value as the index key here. There would likely be problems if the user supplied value matched an index key we define such as `all`, `pending`, or `pending-triggered-named-foo`. Please prepend a prefix like `job-queue-name` to prevent this.

Reply:
Since this was using another indexer I don't think this would be a problem? But either way this is no longer the case as of https://github.com/kubernetes/test-infra/pull/25674/commits/55d9a8ef93e1eccd103b77a00f96d7a84cf0cbbd (this function is removed).